src/components/Card.js

```diff
@@ -13,36 +25,86 @@ export default class Card {
     return cardTemplate;
   }
 
+  // old code, use this first
+
+  // _toggleLike = () => {
+  //   // Do some API function to handle like and unlike
+  //   // After API call, update this._likes
+  //   // Update like count and like button state
+  //   this._likeButton.classList.toggle("card__like-button-active");
+  // };
+
+  // updated code, not sure.
   _toggleLike = () => {
-    this._likeButton.classList.toggle("card__like-button-active");
+    // check if an of the likes belong to you
+    const isLiked = this._likeButton.classList.contains(
+      "card__like-button-active"
+    );
+    this._isLiked = isLiked;
+    this._handleLikeCard(this);
   };
 
+  updateLikes(likes) {
+    // console.log("update like", likes.length);
+    this._cardCountEl.textContent = likes.length;
+    // console.log(likes, this._cardData.likes);
+    this._cardData.likes = likes;
+    this._renderLikes();
+  }
+
+  isLiked() {
+    // console.log(this._cardData.likes);
+    for (let i = 0; i < this._cardData.likes.length; i++) {
```

Comment:
You can use `Array.some` method here `return this._cardData.likes.some(like => like._id === this._userId)`